lr-cli/cortx_setup/commands/storage/config.py

```diff
@@ -210,30 +238,88 @@ def run(self, **kwargs):
+
 
         if setup_type == "VM":
-            if name or storage_type or controller_type:
+            if (
+                name
+                or storage_type
+                or controller_type
+                or cvg
+                or data_devices
+                or metadata_devices
+            ):
                 if not self.enclosure_id:
                     self.enclosure_id = "enc_" + self.machine_id
                     self.refresh_key_map()
                     self.store_in_file()
-                    self.update_pillar_and_conf('enclosure_id', self.enclosure_id)
+                    self.update_pillar_and_conf(
+                        'enclosure_id',
+                        self.enclosure_id
+                    )
 
                 if name:
                     self.update_pillar_and_conf('name', name)
 
-                if storage_type:
+                if storage_type != "virtual":
+                    self.logger.error(
+                        "Storage type needs to be 'virtual' for VM")
+                    raise ValueError("Incorrect argument value provided")
+                else:
                     self.update_pillar_and_conf('storage_type', storage_type)
 
-                if controller_type:
-                    self.update_pillar_and_conf('controller_type', controller_type)
+                if controller_type != "virtual":
+                    self.logger.error(
+                        "Controller type needs to be 'virtual' for VM")
+                    raise ValueError("Incorrect argument value provided")
+                else:
+                    self.update_pillar_and_conf(
+                        'controller_type',
+                        controller_type
+                    )
+
+                if cvg > 1:
+                    self.logger.error(
+                        "The value of CVG has to be either 0 or 1")
+                    raise ValueError("Incorrect argument value provided")
+                else:
+                    self.update_pillar_and_conf('cvg', cvg)
+
+                if data_devices:
+                    ddevices = data_devices.split(",")
+                    for device in ddevices:
+                        cmd_run(f"ls {device}", targets=node_id)
+
+                    self.update_pillar_and_conf('data_devices', ddevices)
+
+                if metadata_devices:
+                    mdevices = data_devices.split(",")
+                    for device in mdevices:
+                        cmd_run(f"ls {device}", targets=node_id)
+
+                    self.update_pillar_and_conf('metadata_devices', mdevices)
+
             else:
-                self.logger.error("Please provide values for name, type and controller_type")
+                self.logger.error(
+                    "Please provide values for name, type and controller_type,"
+                    " cvg, data_devices, metadata_devices"
+                )
```

Comment:
It is `data_devices` here and `data-devices` in actual param. 
Also, for values which are optional and can already be got from Confstore, we must not make it mandatory for the user to provide.

Reply:
yes, it was a typo there.. addressed.